### PR TITLE
test: Benchmark to be run on codspeed

### PR DESCRIPTION
**Pull request by DeaMariaLeon**
## What type of PR is this? (check all applicable)

- [ ] 💾 Refactor
- [ ] 🪄 Feature
- [ ] 🐞 Bug Fix
- [ ] 🔧 Optimization
- [ ] 📚 Documentation
- [x] 🧪 Test
- [ ] 🛠️ Other

## Related issues

- Related issue #
- Closes #

## Checklist

- [x] Code follows style guide
- [ ] Tests added
- [ ] Documented the changes

***

## Please explain your changes below.
This is @hameerabbasi code (shared on slack).

### CODE REVIEW

File: benchmarks/test_benchmark_coo.py [hunk 1]
```diff
@@ -1,15 +1,33 @@
...
+@pytest.mark.parametrize("side", [100, 500, 1000])
+@pytest.mark.parametrize("rank", [1, 2, 3, 4])
+def elemwise_args(side: int, rank: int):
+    if side**rank >= 2**26:
+        pytest.skip()
+    rng = np.random.default_rng(seed=SEED)
+    shape = (side,) * rank
+    x = sparse.random(shape, density=DENSITY, random_state=rng)
+    y = sparse.random(shape, density=DENSITY, random_state=rng)
+    return x, y
```
Comment (hameerabbasi) [inline benchmarks/test_benchmark_coo.py:23]:
```suggestion
@pytest.fixture(scope="module", side=[100, 500, 1000], rank=[1, 2, 3, 4])
def elemwise_args(request):
    side, rank = request.side, request.rank
    if side**rank >= 2**26:
        pytest.skip()
    rng = np.random.default_rng(seed=SEED)
    shape = (side,) * rank
    x = sparse.random(shape, density=DENSITY, random_state=rng)
    y = sparse.random(shape, density=DENSITY, random_state=rng)
    return x, y
```

**Review by hameerabbasi — CHANGES_REQUESTED** [review]
This should fix the PyTest error at least.

File: benchmarks/test_benchmark_coo.py [hunk 2]
```diff
@@ -1,15 +1,32 @@
...
 
+import pytest
+
 import numpy as np
 
+DENSITY = 0.01
+SEED = 42
+
+
+@pytest.fixture(scope="module", side=[100, 500, 1000], rank=[1, 2, 3, 4])
```
Comment (hameerabbasi) [inline benchmarks/test_benchmark_coo.py:13]:
```suggestion
@pytest.fixture(scope="module", params=itertools.product([100, 500, 1000], [1, 2, 3, 4]))
```

File: benchmarks/test_benchmark_coo.py [hunk 3]
```diff
@@ -1,15 +1,32 @@
...
+def elemwise_args(request):
+    side, rank = request.side, request.rank
```
Comment (hameerabbasi) [inline benchmarks/test_benchmark_coo.py:15]:
```suggestion
    side, rank = request.params
```

**Review by hameerabbasi — CHANGES_REQUESTED** [review]
This should fix it.

File: benchmarks/test_benchmark_coo.py [hunk 4]
```diff
@@ -1,15 +1,35 @@
...
+DENSITY = 0.01
+SEED = 42
+
+
+import itertools
+
+
+@pytest.fixture(scope="module", params=itertools.product([100, 500, 1000], [1, 2, 3, 4]))
+def elemwise_args(request):
+    side, rank = request.params
```
Comment (hameerabbasi) [inline benchmarks/test_benchmark_coo.py:18]:
```suggestion
    side, rank = request.param
```

**Review by hameerabbasi — APPROVED** [review]

**Review by DeaMariaLeon — COMMENTED** [review]
I can't approve, but LGTM. :-)

**Pull request merged**